src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationAndGeneratorDriverTranslationAction_Actions.cs

```diff
@@ -103,7 +104,7 @@ internal sealed class AddDocumentsAction(ImmutableArray<DocumentState> documents
             {
                 public override async Task<Compilation> TransformCompilationAsync(Compilation oldCompilation, CancellationToken cancellationToken)
                 {
-                    var syntaxTrees = new List<SyntaxTree>(capacity: documents.Length);
+                    using var _ = ArrayBuilder<SyntaxTree>.GetInstance(documents.Length, out var syntaxTrees);
```

Comment:
>ArrayBuilder<SyntaxTree> [](http://example.com/codeflow?start=34&length=24)

might as well change the other ones in this file too

Reply:
done.